src/Microsoft.VisualStudio.Threading.Analyzers.CSharp/VSTHRD003UseJtfRunAsyncAnalyzer.cs

```diff
@@ -71,6 +71,30 @@ public override void Initialize(AnalysisContext context)
         context.RegisterSyntaxNodeAction(Utils.DebuggableWrapper(this.AnalyzeLambdaExpression), SyntaxKind.ParenthesizedLambdaExpression);
     }
 
+    private static bool IsSymbolAlwaysOkToAwait(ISymbol? symbol)
+    {
+        if (symbol is IFieldSymbol field)
+        {
+            // Allow the TplExtensions.CompletedTask and related fields.
+            if (field.ContainingType.Name == Types.TplExtensions.TypeName && field.BelongsToNamespace(Types.TplExtensions.Namespace) &&
```

Comment:
both this and the conditional on 88 are a little bit difficult to grok based on initial read. Consider making a method with a descriptive name to make it easier to understand at a glance.

Reply:
I agree it is. I have a code comment above that explains it, and this _is_ the method that keeps things simple for the caller. :)